### PR TITLE
Feat: Feature set II

### PR DESCRIPTION
**Pull request by GK72**

### CODE REVIEW

File: baldr.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# A marker indicating the source.
```
Comment (lafette21) [inline baldr.yaml:1]: Debug purpose

File: src/main.rs [hunk 2]
```diff
@@ -72,64 +73,78 @@ fn create_compile_cmd_symlink(src: &PathBuf, dst: &PathBuf) -> Result<(), io::Er
...
     match exes.len() {
         1 => {
-            let mut cmd = Command::new(&exes[0]);
+            let mut cmd = (|| -> Result<Command, String> {
+                if args.debug {
+                    let debugger = config.get::<String>("debugger")
+                        .map_err(|e| format!("No debugger is configured: {e}"))?;
+
+                    let mut cmd = Command::new(debugger);
+                    cmd.arg("--args");
```
Comment (lafette21) [inline src/main.rs:99]: This is specific to gdb.

**Review by lafette21 — APPROVED** [review]

File: src/main.rs (hunk 2 above)
Comment (GK72) [reply src/main.rs]: Not anymore.

File: src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+
+/// Invoke CMake's configure command.
+///
+/// # Errors
+///
+/// Returns an error, if the process cannot be started.
+pub fn configure(path: &Path, args: &Args, config: &Config) -> Result<ExitStatus, String> {
+    let mut cmd = Command::new("cmake");
+
+    let cc: String = config.get("compiler.cc").unwrap_or_default();
```
Comment (lafette21) [inline src/lib.rs:206]: get_cc?

File: src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+    let cxx: String = config.get("compiler.cxx").unwrap_or_default();
```
Comment (lafette21) [inline src/lib.rs:207]: get_cxx?

**Review by lafette21 — APPROVED** [review]

**Pull request merged**